stripped/colabs/agregacje_kilkuletnie_stripped.py

```diff
@@ -0,0 +1,669 @@
+
+pd_2023_monthly_avg_4.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/4years/NASA_monthly_avg_2023.csv')
+
+# 2022
+nasa_2022_monthly_avg_4 = result_df_4.filter(f'Year = {2022}')
+pd_2022_monthly_avg_4 = nasa_2022_monthly_avg_4.toPandas()
+
+pd_2022_monthly_avg_4.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/4years/NASA_monthly_avg_2022.csv')
+
+# 2020
+nasa_2020_monthly_avg_4 = result_df_4.filter(f'Year = {2020}')
+pd_2020_monthly_avg_4 = nasa_2020_monthly_avg_4.toPandas()
+
+pd_2020_monthly_avg_4.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/4years/NASA_monthly_avg_2020.csv')
+
+# 2000
+nasa_2000_monthly_avg_4 = result_df_4.filter(f'Year = {2000}')
+pd_2000_monthly_avg_4 = nasa_2000_monthly_avg_4.toPandas()
+
+pd_2000_monthly_avg_4.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/4years/NASA_monthly_avg_2000.csv')
+
+# 1979
+nasa_1979_monthly_avg_4 = result_df_4.filter(f'Year = {1979}')
+pd_1979_monthly_avg_4 = nasa_1979_monthly_avg_4.toPandas()
+
+pd_1979_monthly_avg_4.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/4years/NASA_monthly_avg_1979.csv')
+
+"""n=5:"""
+
+result_df_5 = overYearMonthStats(columns_to_avg, n=5)
+result_df_5.show(15)
+
+# 2023
+nasa_2023_monthly_avg_5 = result_df_5.filter(f'Year = {2023}')
+pd_2023_monthly_avg_5 = nasa_2023_monthly_avg_5.toPandas()
+
+pd_2023_monthly_avg_5.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/5years/NASA_monthly_avg_2023.csv')
+
+# 2022
+nasa_2022_monthly_avg_5 = result_df_5.filter(f'Year = {2022}')
+pd_2022_monthly_avg_5 = nasa_2022_monthly_avg_5.toPandas()
+
+pd_2022_monthly_avg_5.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/5years/NASA_monthly_avg_2022.csv')
+
+# 2020
+nasa_2020_monthly_avg_5 = result_df_5.filter(f'Year = {2020}')
+pd_2020_monthly_avg_5 = nasa_2020_monthly_avg_5.toPandas()
+
+pd_2020_monthly_avg_5.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/5years/NASA_monthly_avg_2020.csv')
+
+# 2000
+nasa_2000_monthly_avg_5 = result_df_5.filter(f'Year = {2000}')
+pd_2000_monthly_avg_5 = nasa_2000_monthly_avg_5.toPandas()
+
+pd_2000_monthly_avg_5.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/5years/NASA_monthly_avg_2000.csv')
+
+# 1979
+nasa_1979_monthly_avg_5 = result_df_5.filter(f'Year = {1979}')
+pd_1979_monthly_avg_5 = nasa_1979_monthly_avg_5.toPandas()
+
+pd_1979_monthly_avg_5.to_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/5years/NASA_monthly_avg_1979.csv')
+
+"""# Dane do modeli
+
+Zbiór anotowany:
+"""
+
+NASA_sample_an = pd.read_csv('/content/drive/MyDrive/BigMess/NASA/NASA_an.csv', sep=';')
+
+NASA_sample_an
+
+"""## Średnie 3-letnie"""
+
+pd_2023_monthly_avg_3 = pd.read_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/3years/NASA_monthly_avg_2023.csv')
+del pd_2023_monthly_avg_3[pd_2023_monthly_avg_3.columns[0]]
+
+monthly_avg_3_july2023 = pd_2023_monthly_avg_3[pd_2023_monthly_avg_3['Month'] == 7]
+
+monthly_avg_3_july2023
```

Comment:
Wydaje mi się, że nie jest potrzebne wyświetlanie całego DataFramu (od tej linijki wyświetlasz 9 pełnych zbiorów danych). Według mnie jak bardzo chcesz je wyświetlić użyj funkcji head() lub wcale ich nie wyświetlaj jeżeli nie będziesz o nich mówić. Zamiast tego można na jakimś wykresie porównać różnice między średnimi z 3, 4 i 5 lat.

Reply:
Dobra uwaga, w colabie wyświetlało mi się tylko jakieś 20 wierszy, więc nie przywiązałam do tego szczególnej wagi. Już usunęłam to wyświetlanie, a zamiast tego, tak jak zasugerowałeś, dodałam dla każdej zmiennej wykresy słupkowe dla 10 losowych lokacji

Reply:
Dla nas może to mieć niewielkie znaczenie, ale dobra praktyka mówi by w takich sytuacjach wyświetlać sample. Niby jak head, ale zawsze zagląda dalej niż na pierwsze kilka wierszy.